### PR TITLE
Fix particle shader deterministic random values

### LINKED ISSUE #78771: Particles change when they are fully stopped
### Godot version

v4.1.rc1.official [1f9e540f1]

### System information

Godot v4.1.rc1 - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 960 (NVIDIA; 31.0.15.3623) - AMD FX(tm)-6300 Six-Core Processor (6 Threads)

### Issue description

When damping is used and particle is fully stopped, it gets new shape, color, angular velocity, etc. It looks like it's being recreated from scratch.

### Steps to reproduce

1. Run the project.
2. Open block_hit_particles.tscn.

### Minimal reproduction project

[Bugged Particles.zip](https://github.com/godotengine/godot/files/11887286/Bugged.Particles.zip)

### LINKED ISSUE #80625: In the particle shader the deterministic sequence of random values is not guaranteed
### Godot version

v4.1.1.stable.official [bd6af8e0e]

### System information

Windows 10.0.22621 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1650 SUPER (NVIDIA; 31.0.15.3130) - Intel(R) Core(TM) i5-10400F CPU @ 2.90GHz (12 Threads)

### Issue description

I think I found a bug in the particle shader. It revolves around the following line:

```
// apply linear acceleration
force += length(VELOCITY) > 0.0 ? normalize(VELOCITY) * tex_linear_accel * mix(linear_accel_min, linear_accel_max, rand_from_seed(alt_seed)) : vec3(0.0);

```
As long as VELOCITY is greater than 0, "rand_from_seed(alt_seed)" is called among other things. Thereby "alt_seed" is changed.
But if now damping is active, VELOCITY reaches 0 at some point. Then "rand_from_seed(alt_seed)" is NOT called anymore.
That means: all following random values, which depend on "rand_from_seed()", change once abruptly.

### Steps to reproduce

The error probably manifests itself in very different ways. As soon as damping brings the speed to 0, there will be one-time jumpy changes in all aspects that use edge_fom_seed:

- radial acceleration
- tangential acceleration
- orbit velocity
- damping
- angular velocity
- animation offset and speed
- hue

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by sepTN**
Fixes #80625 

_bugsquad edit: Fixes https://github.com/godotengine/godot/issues/78771_

### CODE REVIEW

**Comment by AThousandShips** [comment]
There are more cases in the same file just a few lines down

File: scene/resources/particle_process_material.cpp [hunk 1]
```diff
@@ -683,11 +683,13 @@ void ParticleProcessMaterial::_update_shader() {
 		code += "	pos.z = 0.0;\n";
 	}
 	code += "	// apply linear acceleration\n";
-	code += "	force += length(VELOCITY) > 0.0 ? normalize(VELOCITY) * tex_linear_accel * mix(linear_accel_min, linear_accel_max, rand_from_seed(alt_seed)) : vec3(0.0);\n";
+	code += "	vec3 linear_accel_rand = mix(linear_accel_min, linear_accel_max, rand_from_seed(alt_seed));\n";
+	code += "	force += length(VELOCITY) > 0.0 ? normalize(VELOCITY) * tex_linear_accel * linear_accel_rand : vec3(0.0);\n";
```
Comment (AThousandShips) [inline scene/resources/particle_process_material.cpp:687]:
```suggestion
	code += "	float linear_accel_rand = rand_from_seed(alt_seed);\n";
	code += "	force += length(VELOCITY) > 0.0 ? normalize(VELOCITY) * tex_linear_accel * mix(linear_accel_min, linear_accel_max, linear_accel_rand) : vec3(0.0);\n";
```
To do the least processing in that branch

File: scene/resources/particle_process_material.cpp [hunk 2]
```diff
@@ -683,11 +683,13 @@ void ParticleProcessMaterial::_update_shader() {
...
 	code += "	// apply radial acceleration\n";
 	code += "	vec3 org = EMISSION_TRANSFORM[3].xyz;\n";
 	code += "	vec3 diff = pos - org;\n";
-	code += "	force += length(diff) > 0.0 ? normalize(diff) * tex_radial_accel * mix(radial_accel_min, radial_accel_max, rand_from_seed(alt_seed)) : vec3(0.0);\n";
+	code += "	vec3 radial_accel_rand = mix(radial_accel_min, radial_accel_max, rand_from_seed(alt_seed));\n";
+	code += "	force += length(diff) > 0.0 ? normalize(diff) * tex_radial_accel * radial_accel_rand : vec3(0.0);\n";
```
Comment (AThousandShips) [inline scene/resources/particle_process_material.cpp:692]:
```suggestion
	code += "	float radial_accel_rand = rand_from_seed(alt_seed);\n";
	code += "	force += length(diff) > 0.0 ? normalize(diff) * tex_radial_accel * mix(radial_accel_min, radial_accel_max, radial_accel_rand) : vec3(0.0);\n";
```
Same

**Review by clayjohn — APPROVED** [review]
Looks good to me. Tested locally and can confirm that this fixes https://github.com/godotengine/godot/issues/78771

**Comment by akien-mga** [comment]
Thanks!

**Comment by YuriSizov** [comment]
Cherry-picked for 4.1.3.

**Pull request merged**